### PR TITLE
doc: Add pull request metrics information

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
+An approximation of the time a change takes to be developed.
+This metric assumes you are not developing after the PR is open.
+
+This metric is the difference between the pull request base commit author timestamp and the pull request created timestamp.
+
+#### Time to Review
+
+The time since a pull request is opened until it is merged.
+
+This metric is the difference between the the pull request merged timestamp and the pull request created timestamp.
```
Comment (mrfyda) [inline docs/metrics.md:66]:
```suggestion
This metric is the difference between the pull request merged timestamp and the pull request created timestamp.
```

File: docs/metrics.md [hunk 2]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
 | Elite                 | Less than 1 day              |
 
+### Sub-metrics
+
+There are several metrics that directly influence the lead time for changes,
+and can help you track in more detail what needs to be improved in your workflow.
+All the sub-metrics are aggregated by an interval (day, week or month) and have values for
+the average, median, and percentiles 5%, 20%, 25%, 75%, 80%, 95%.
+
+These metrics are extracted from pull requests,
```
Comment (mrfyda) [inline docs/metrics.md:51]: consider using a list to be more clear

File: docs/metrics.md [hunk 3]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
+only include pull requests that have been merged (ignores closed without merge and currently open) and
+draft pull requests are treated as normal pull requests.
+
+#### Time to Open
+
+An approximation of the time a change takes to be developed.
+This metric assumes you are not developing after the PR is open.
```
Comment (mrfyda) [inline docs/metrics.md:58]: assumes but takes those PRs into account

File: docs/metrics.md [hunk 4]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
+
+This metric is the difference between the pull request base commit author timestamp and the pull request created timestamp.
```
Comment (mrfyda) [inline docs/metrics.md:60]:
```suggestion
This metric is calculated as the time between the first commit in the pull request's branch and the pull request being created.
```

File: docs/metrics.md [hunk 5]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
+
+#### Time to Review
+
+The time since a pull request is opened until it is merged.
```
Comment (mrfyda) [inline docs/metrics.md:64]:
```suggestion
The time spent in the code review process. Once a PR is opened, how long does it take to merge?
```

File: docs/metrics.md [hunk 6]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
+All the sub-metrics are aggregated by an interval (day, week or month) and have values for
+the average, median, and percentiles 5%, 20%, 25%, 75%, 80%, 95%.
+
+These metrics are extracted from pull requests,
+only include pull requests that have been merged (ignores closed without merge and currently open) and
+draft pull requests are treated as normal pull requests.
+
+#### Time to Open
+
+An approximation of the time a change takes to be developed.
```
Comment (mrfyda) [inline docs/metrics.md:57]:
```suggestion
An approximation of the time a change takes to be developed. How long does it take features to be developed?
```

File: docs/metrics.md (hunk 3 above)
Comment (rtfpessoa) [reply docs/metrics.md:58]: Since we stopped calling time in development this probably makes no sense

Comment (rtfpessoa) [reply docs/metrics.md:58]:
```suggestion
```

File: docs/metrics.md [hunk 7]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
+### Sub-metrics
+
+There are several metrics that directly influence the lead time for changes,
+and can help you track in more detail what needs to be improved in your workflow.
+All the sub-metrics are aggregated by an interval (day, week or month) and have values for
+the average, median, and percentiles 5%, 20%, 25%, 75%, 80%, 95%.
+
+These metrics are extracted from pull requests,
+only include pull requests that have been merged (ignores closed without merge and currently open) and
+draft pull requests are treated as normal pull requests.
```
Comment (rtfpessoa) [inline docs/metrics.md:53]:
```suggestion
draft pull requests are treated as normal pull requests.
Also excludes pull requests without commits or where the first commit was authored after the pull request was open.
```

Comment (mrfyda) [inline docs/metrics.md:53]:
```suggestion
These metrics are based on pull requests.
They:

- Include Draft pull requests
- Exclude declined pull requests
- Exclude open pull requests
```

File: docs/metrics.md (hunk 3 above)
Comment (mrfyda) [reply docs/metrics.md:58]: other tools solve this problem by assuming that commits after the PR is opened is rework, and have different metrics for that.
```suggestion

```

File: docs/metrics.md [hunk 8]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
 | High                  | Between 1 day and 1 week     |
 | Elite                 | Less than 1 day              |
 
+### Sub-metrics
+
+There are several metrics that directly influence the lead time for changes,
```
Comment (mrfyda) [inline docs/metrics.md:46]:
```suggestion
There are several metrics that directly influence **Lead time for changes**,
```

File: docs/metrics.md [hunk 9]
```diff
@@ -42,6 +41,29 @@ When we aggregate Lead time for changes we use the maximum of a period of time.
...
+and can help you track in more detail what needs to be improved in your workflow.
+All the sub-metrics are aggregated by an interval (day, week or month) and have values for
+the average, median, and percentiles 5%, 20%, 25%, 75%, 80%, 95%.
+
+These metrics are extracted from pull requests,
+only include pull requests that have been merged (ignores closed without merge and currently open) and
+draft pull requests are treated as normal pull requests.
+
+#### Time to Open
```
Comment (mrfyda) [inline docs/metrics.md:55]: maybe a table makes more sense for this type of information (kind of reference)

Comment (rtfpessoa) [reply docs/metrics.md:60]: hmm, not sure what you mean

Comment (mrfyda) [reply docs/metrics.md:60]: something like 👇 
 metric name | metric description| metric formula | units 
-|-|-|-
time to open | ...
time in review | ...

Comment (rtfpessoa) [reply docs/metrics.md:60]: we can try that if we think it looks better

File: docs/metrics.md (hunk 1 above)
Comment (pauloribeiro-codacy) [reply docs/metrics.md]: Here it's the time between creating and **merging** the pull request.

**Review by pauloribeiro-codacy — APPROVED** [review]
If the metrics **Time to open** and **Time to review** are the new ones available now, we can already merge this pull request :rocket: and I'll do some more iterations on top to improve the overall metrics page. :+1:

**Pull request merged**